src/main/java/seedu/address/logic/commands/AddTagCommand.java

```diff
@@ -0,0 +1,206 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Add new tag(s) for an existing person
+ */
+public class AddTagCommand extends Command {
+    public static final String COMMAND_WORD = "addTag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + "Add tag(s) for the person identified "
+            + "by the index number used in the displayed person list "
+            + "Parameters: INDEX (must be a positive interger) "
+            + "[" + PREFIX_TAG + "TAG]...\n";
+
+    public static final String MESSAGE_ADD_TAG_SUCCESS = "Tag added: %1$s";
+    public static final String MESSAGE_NOT_ADD = "At least one tag to be provided.";
+    private final Index index;
+    private final AddTagCommand.AddTagDescriptor addTagDescriptor;
+
+    /**
+     * @param index of the person in the filtered person list to edit
+     * @param addTagDescriptor details to edit the person with
+     */
+    public AddTagCommand(Index index, AddTagCommand.AddTagDescriptor addTagDescriptor) {
+        requireNonNull(index);
+        requireNonNull(addTagDescriptor);
+
+        this.index = index;
+        this.addTagDescriptor = new AddTagCommand.AddTagDescriptor(addTagDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = createEditedPerson(personToEdit, addTagDescriptor);
```

Comment:
It may not be necessary to use `addTagDescriptor` to create an edited person since only tag list is changed. This might make the logic to complicated. Does anyone has any idea on this part?

Reply:
Yes because it replace a new Person object. Right ?

Reply:
I am not sure whether it will be better if we can handle the process use another function inside the `modelManager` class.  But your implementation definitely works. We can discuss about this in the next stage.